### PR TITLE
feat(Timeline): Convert Timeline to CSS modules behind team feature flag

### PR DESCRIPTION
**Pull request by francinelucca**
Closes https://github.com/github/primer/issues/4366

#### New

- `Timeline` dev story to test sx and style props with VRT
- `Timeline` module css
- Snapshots for new dev story
- Tests for `Timeline.Break` and `Timeline.Body`
- `className` support tests for all `Timeline` components

#### Changed

- Refactor `Timeline` e2e tests and add new dev story
- Refactor `Timeline` to use CSS modules when team feature flag is turned on
- Deprecate `TimelineItemsProps` in favor of `TimelineItemProps`

### Rollout strategy

- [ ] Patch release
- [x] Minor release
- [ ] Major release; if selected, include a written rollout or migration plan
- [ ] None; if selected, include a brief description as to why

### Merge checklist

- [x] Added/updated tests
- [ ] Added/updated documentation
- [x] Added/updated previews (Storybook)
- [ ] Changes are [SSR compatible](https://github.com/primer/react/blob/main/contributor-docs/CONTRIBUTING.md#ssr-compatibility)
- [x] Tested in Chrome
- [ ] Tested in Firefox
- [ ] Tested in Safari
- [ ] Tested in Edge
- [ ] (GitHub staff only) Integration tests pass at github/github ([Learn more about how to run integration tests](https://github.com/github/primer-engineering/blob/main/how-we-work/testing-primer-react-pr-at-dotcom.md))

### CODE REVIEW

File: packages/react/src/Timeline/Timeline.module.css [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    height: 32px;
+    margin-right: var(--base-size-8);
+    /* stylelint-disable-next-line primer/spacing */
+    margin-left: -15px;
+    flex-shrink: 0;
+    overflow: hidden;
+    color: var(--fgColor-muted);
+    
+    /* TODOl not quite sure if this is the correct migration for this line */
+    background-color: var(--timelineBadge-bgColor);
```
Comment (francinelucca) [inline packages/react/src/Timeline/Timeline.module.css:71]: Can I get a check on this line? I wasn't quite sure what the correct mapping was here. i pulled the `--timelineBadge-bgColor` from the computer styles in production

File: packages/react/src/Timeline/__tests__/Timeline.test.tsx [hunk 2]
```diff
@@ -49,4 +88,80 @@ describe('Timeline.Badge', () => {
...
+            primer_react_css_modules_ga: true,
+          }}
+        >
+          <Element />
+        </FeatureFlags>
+      )
+    }
+    expect(HTMLRender(<FeatureFlagElement />).container.firstChild?.firstChild).toHaveClass('test-class-name')
+  })
+})
```
Comment (francinelucca) [inline packages/react/src/Timeline/__tests__/Timeline.test.tsx:109]: The sx-compliant Timeline.Badge doesn't support className, I thought it'd make sense to add it to the css modules refactor and therefore I'm only testing for that one here, but lmk if that doesn't make sense and I can revert!

File: packages/react/src/Timeline/Timeline.tsx [hunk 3]
```diff
@@ -1,76 +1,158 @@
...
+    ${sx};
+  `,
+)
+
+/**
+ * @deprecated Use the `TimelineItemProps` type instead
+ */
+export type TimelineItemsProps = StyledTimelineItemProps & HTMLProps<HTMLDivElement>
+
+export type TimelineItemProps = StyledTimelineItemProps & HTMLProps<HTMLDivElement>
```
Comment (francinelucca) [inline packages/react/src/Timeline/Timeline.tsx:114]: It seems to me the "correct" prop name would be `TimelineItemProps` singular, so I'm proposing deprecating `TimelineItemsProps` in favor of `TimelineItemProps`. Would love to hear thoughts on this one

**Review by jonrohan — APPROVED** [review]

File: packages/react/src/Timeline/Timeline.module.css [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+  overflow: hidden;
+  color: var(--fgColor-muted);
+
+  /* TODOl not quite sure if this is the correct migration for this line */
+  background-color: var(--timelineBadge-bgColor);
+  /* stylelint-disable-next-line primer/colors */
+  border-color: var(--bgColor-default);
+  border-style: solid;
+  /* stylelint-disable-next-line primer/borders */
+  border-width: 2px;
```
Comment (langermank) [inline packages/react/src/Timeline/Timeline.module.css:75]:
```suggestion
  border-width: var(--borderWidth-thick);
```

File: packages/react/src/Timeline/Timeline.module.css [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+  margin-top: var(--base-size-4);
+  font-size: var(--text-body-size-medium);
+  color: var(--fgColor-muted);
+  flex: auto;
+}
+
+.TimelineBreak {
+  position: relative;
+  z-index: 1;
+  height: 24px;
```
Comment (langermank) [inline packages/react/src/Timeline/Timeline.module.css:93]:
```suggestion
  height: var(--base-size-24);
```

File: packages/react/src/Timeline/Timeline.module.css [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+  margin: 0;
+  margin-bottom: calc(-1 * var(--base-size-16));
+  margin-left: 0;
+  background-color: var(--bgColor-default);
+  border: 0;
+  /* stylelint-disable-next-line primer/borders */
+  border-top: var(--base-size-4) solid var(--borderColor-default);
```
Comment (langermank) [inline packages/react/src/Timeline/Timeline.module.css:100]:
```suggestion
  border-top: var(--borderWidth-thicker) solid var(--borderColor-default);
```

**Review by langermank — APPROVED** [review]
Just a few nitpicks! The color vars look good 👍

**Pull request merged**